remi/server.py

```diff
@@ -147,10 +147,15 @@ def handle(self):
             else:
                 if not self.read_next_message():
                     k = get_instance_key(self)
-                    clients[k].websockets.remove(self)
-                    self.handshake_done = False
-                    self._log.debug('ws ending websocket service')
-                    break
+                    try:
+                        clients[k].websockets.remove(self)
```

Comment:
Hello @dirkcgrunwald ! 
Looking at changes you made, I suppose that beside others, these 8 grouped lines (from 150 to 158) are the necessary part to be C9 compatible. Others are refinements. Is this correct?

Reply:
There are 2-3 others parts that are needed.

In      if 'C9_HOSTNAME' in os.environ:
 +            key = data.decode().split('sec-websocket-key: ')[1].split('\r\n')[0]
 +        else:
 +            key = data.decode().split('Sec-WebSocket-Key: ')[1].split('\r\n')[0]

it appears that the C9 proxy makes the response headers in lower case. It's also possible to use a try...except to try the upper case, fail and then revert to lower case.

In addition, in the default setup, if you use '0.0.0.0' as the host IP, remi changes this to 127.0.0.1, which isn't proxied. The user doesn't know the IP address of their server because of the proxy, so they need to leave '0.0.0.0' as an option. I had guarded those changes with checks for the C9 environment.

This may all be moot since C9 is being changed to AWS Cloud and I think those don't use the proxy.